### PR TITLE
[node-manager] capi-controller-manager linter fix

### PR DESCRIPTION
**Pull request by borg-z**
## Description

Several changes to RBAC and enables the linter for CAPI.

- In the roles `capi-controller-manager` and `capi-cloud-cluster-controller-manager`, the wildcard (`*`) has been replaced with a list of specific resources defined in the CRD.
- unused `capi-cloud-cluster-controller-manager` `ServiceAccount` has been removed from several roles.

## Why do we need it, and what problem does it solve?

Enable Linter for CAPI and security improvements 

## What is the expected result?

Changing some Roles as described in the descriptipon.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: fix
summary: Fixed several RBAC resources in the node-manager module.
impact: 
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/vcd'...

File: modules/040-node-manager/templates/capi-controller-manager/rbac-for-us.yaml [hunk 1]
```diff
@@ -85,20 +93,18 @@ rules:
...
+      - kubeadmcontrolplanes
+      - kubeadmcontrolplanetemplates
+      - deckhousecontrolplanes
+      - staticclusters
+      - staticmachines
+      - staticmachinetemplates
+      - vcdclusters
+      - vcdclustertemplates
+      - vcdmachines
+      - vcdmachinetemplates
```
Comment (name212) [inline modules/040-node-manager/templates/capi-controller-manager/rbac-for-us.yaml:107]: need to add zvirt resources

**Review by name212 — CHANGES_REQUESTED** [review]
one thing

**Review by name212 — APPROVED** [review]

**Review by RomanenkoDenys — COMMENTED** [review]
one question

File: modules/040-node-manager/templates/capi-controller-manager/rbac-for-us.yaml [hunk 2]
```diff
@@ -449,9 +452,6 @@ subjects:
     namespace: d8-cloud-instance-manager
   - kind: User
     name: capi-controller-manager
-  - kind: ServiceAccount
-    name: capi-cloud-cluster-controller-manager
-    namespace: d8-cloud-instance-manager
   - kind: User
     name: capi-cloud-cluster-controller-manager
 {{- end }}
```
[the remark this answers is not in the document]
Comment (borg-z) [reply modules/040-node-manager/templates/capi-controller-manager/rbac-for-us.yaml]: I don't know, some kind of old artifact

**Review by z9r5 — APPROVED** [review]

**Pull request merged**